src/rp2_common/hardware_flash/flash.c

```diff
@@ -370,3 +406,155 @@ void flash_devinfo_set_cs_gpio(uint cs, uint gpio) {
 }
 
 #endif // !PICO_RP2040
+//-----------------------------------------------------------------------------
+/**
+ * Low level flash functions are based on:
+ * github.com/raspberrypi/pico-bootrom-rp2040/blob/master/bootrom/program_flash_generic.c,
+ * github.com/raspberrypi/pico-bootrom-rp2350/blob/master/src/main/arm/varm_generic_flash.c
```

Comment:
@kartben 

I just want to confirm.

The code referenced here is the same BSD-3 as this code
and was created by `Raspberry Pi (Trading) Ltd.`, the same author as the original of this code.

Since they're both BSD-3, I think this description is sufficient, but what do you think?